### PR TITLE
add release drafter

### PR DESCRIPTION
**Pull request by casz**
Let the automation begin! 🙌

### CODE REVIEW

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# https://github.com/jenkinsci/.github/blob/master/.github/release-drafter.adoc
+_extends: .github
+version-template: $MAJOR.$MINOR.$PATCH
+tag-template: hashicorp-vault-$NEXT_PATCH_VERSION
```
Comment (baymac) [inline .github/release-drafter.yml]: Shouldn't this be `gitlab-branch-source-$NEXT_PATCH_VERSION`? @casz

Comment (casz) [reply .github/release-drafter.yml]: I thought I updated it 😭

Comment (baymac) [reply .github/release-drafter.yml]: It's okay

Comment (baymac) [reply .github/release-drafter.yml]: I think we can do it directly on `master`. And add the commit to this pr as comment.

Comment (casz) [reply .github/release-drafter.yml]: Nah PR, add skip-changelog 😅 
See config here: https://github.com/jenkinsci/.github/blob/master/.github/release-drafter.yml

**Comment by baymac** [comment]
Whoa! That's cool! :D

**Pull request merged**